src/main/java/com/crud/library/repository/RentalRepository.java

```diff
@@ -7,11 +7,9 @@
 @Repository
 public interface RentalRepository extends JpaRepository<Rental, Long> {
 
-    Rental findById(long id);
+    Rental findByUser_IdAndExemplar_Id(long userId, long exemplarId);
 
-    void deleteById(long id);
+    Rental getByExemplar_Id(long exemplarId);
```

Comment:
Zobacz, ta metoda zwróci losowy rekord z tabeli rental, który zawiera egzemplarz.
Zalozmy, ze mamy takie dane w tabelce rentals:
id   |  exemplar_id  | user_id     | rent_date    | return_date
1     |   6                   | 34              | 13-09-2021      |  20-09-2021
2    |   9                    | 56              | 19-09-2021      | null
3    |    6                   | 43              | 29-09-2021     | null

twoja metoda może zwrócić rental z id = 1 (czyli będziesz chciał zwróci książkę exemplarId = 6, którą już użytkownik z userId = 24 zwrócił), a chodzi nam o rental z id = 3. 
findByExemplar_IdOrderByRentDateDesc(long exemplarId) - taka metoda nam zwróci ostanie czasowo wypożyczenie książki, czyli to z id = 3

Reply:
Podobnie z powyższą metodą: findByUser_IdAndExemplar_IdOrderByRentDateDesc(...)